### ISSUE TITLE
devon.bat fails to setup environment on windows.

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon-ide#206]
After successful setup:
```
C:\projects\cool-project\workspaces>devon

C:\projects\cool-project\workspaces>mvn -v
'mvn' is not recognized as an internal or external command,
operable program or batch file.
```

It is incredible how hard it is to test this on all platforms and make it work smooth and perfect.

**Comment by hohwille** [comment]
This is the same error as what I already debugged in powershell: #198 
I have multiple `goto` statements inside `if` statements in `devon.bat` but this one is somehow causing a strange error that I can not trace down:
https://github.com/devonfw/devon-ide/blob/bf849a0ad3115ba5bf7cacc2b42ee810956adfde/scripts/src/main/resources/scripts/devon.bat#L114

I have to dig in the history to see what has changed in this regard.
If we are able to fix this error, I can create a new release and we should finally reach the quality we need for further piloting and bringing `devon-ide` to a wider base of users on our way to the official release.

**Comment by hohwille** [comment]
It is kind of magic:
* [Various fixes and improvements](https://github.com/devonfw/devon-ide/commit/8ce2b41fab0d2b505ece3761e9c3b052af1c99a9#diff-276ddb51568b83cef5cb8c172f37f8aa) - I introduced a new label `devon_bat_installed` for the updating as BAT does not easily provide boolean operations in if conditions.
* [#193: windows fix](https://github.com/devonfw/devon-ide/commit/0af772e6512a9fd05a27f493d4ec7d057f760faa#diff-276ddb51568b83cef5cb8c172f37f8aa) - here I added an extra `goto :eof` before `setup_env` and added a `  set DEVON_IDE_HOME=%CD%` into the `if` now buggy statement. However, outcommenting these two changes with `REM` on the latest code does not make any difference.

**Comment by hohwille** [comment]
Windows CMD/BAT is pure hell.
Changing 
```
  echo devon-ide environment variables have been set for %CD% (workspace %WORKSPACE%)
```
to
```
  echo devon-ide environment variables have been set for %CD%
  REM  (workspace %WORKSPACE%)
```

resolves the error. This used to work for ages. It also only breaks if followed by `popd` and the `goto :eof`. For some undeclared reason this combination freaks CMD out and makes it terminate the script at this point without any human-understandable reason.

**Comment by hohwille** [comment]
Indeed the braces were causing the strange error/effect. Maybe they would need to be escaped maybe with backslashes. At least quotes do not help as they are echoed with the text.
I assume that the braces are calling a sub-shell to execute the statement.
As `workspace %WORKSPACE%` is not a valid statement, this subshell failed and the exit code made the `devon.bat` to exit at this point.
So far this would be a logical explanation apart from the fact that commenting out the `goto :eof` after this statement made the `devon.bat` to proceed and pass the braced echo statement. After dealing with `SETLOCAL EnableDelayedExpansion` (see https://ss64.com/nt/delayedexpansion.html) I learned that it is at some degree a waste of time trying to really understand CMD/BAT. You have to accept that it is a pure pile of hacks that have historically (or hysterically) grown.

**Comment by hohwille** [comment]
Fixed with PR #207
